### PR TITLE
Streamlines the augment_dataloader method in DPO

### PR DESCRIPTION
**Pull request by trias702**
# What does this PR do ?

Streamlines the augment_dataloader code in DPO making it less messy. There is no change to functionality, this is just a code cleanup.

# Changelog 
- Please update the [CHANGELOG.md](/CHANGELOG.md) under next version with high level changes in this PR.

# Before your PR is "Ready for review"
**Pre checks**:
- [X] Make sure you read and followed [Contributor guidelines](/CONTRIBUTING.md)
- [ ] Did you write any new necessary tests?
- [ ] Did you add or update any necessary documentation? Make sure to also update the [NeMo Framework User Guide](https://docs.nvidia.com/nemo-framework/user-guide/latest/index.html) which contains the tutorials

# Checklist when contributing a new algorithm
- [ ] Does the trainer resume and restore model state all states?
- [ ] Does the trainer support all parallelism techniques(PP, TP, DP)?
- [ ] Does the trainer support `max_steps=-1` and `validation`?
- [ ] Does the trainer only call APIs defined in [alignable_interface.py](/nemo_aligner/models/alignable_interface.py)?
- [ ] Does the trainer have proper logging?

# Additional Information
* Related to # (issue)

### CODE REVIEW

File: nemo_aligner/algorithms/dpo.py [hunk 1]
```diff
@@ -308,27 +308,19 @@ def load_state_dict(self, state_dict):
...
                 batch = next(iter_dataloader)
+                logprobs = self.model.get_ref_policy_logprobs(batch).cpu()
+                chosen_logps, reject_logps = torch.split(logprobs, len(logprobs) // 2, dim=0)
+                batch["ref_policy_log_probs_chosen"] = chosen_logps
+                batch["ref_policy_log_probs_rejected"] = reject_logps
+
+                yield batch
+                del logprobs, chosen_logps, reject_logps
             except StopIteration:
                 done = True
```
Comment (odelalleau) [inline nemo_aligner/algorithms/dpo.py:323]: While you're at it, can you get rid of the `done` flag and just `break` here?

**Review by odelalleau — COMMENTED** [review]
lgtm, just a minor suggestion to finish the code simplification

File: nemo_aligner/algorithms/dpo.py (hunk 1 above)
Comment (trias702) [reply nemo_aligner/algorithms/dpo.py]: Done, please have a look

**Review by odelalleau — APPROVED** [review]
lgtm, thanks!

**Pull request merged**